File: Design2_Q1.py
class MyQueue:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.arr=[]


    def push(self, x: int) -> None:
        """
        Push element x to the back of queue.
        """
        self.arr.append(x)


    def pop(self) -> int:
        """
        Removes the element from in front of queue and returns that element.
        """
        if len(self.arr)>0:
            x=self.arr[0]
            self.arr.remove(self.arr[0])
            return x



    def peek(self) -> int:
        """
        Get the front element.
        """
        if len(self.arr)>0:
            return self.arr[0]



    def empty(self) -> bool:
        """
        Returns whether the queue is empty.
        """
        if len(self.arr)>0:
            return False
        return True



# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: design2_q1.cpp
/*
********************Time complexity******************
Push()      - O(1)
Pop()       - O(N)
Peek()      - O(N)
empty()     - O(1)
********************Space complexity*****************
O(N)
*****************************************************
*/
class MyQueue {
    stack<int> s1, s2;
public:
    /** Initialize your data structure here. */
    MyQueue() {
        
    }
    
    /** Push element x to the back of queue. */
    void push(int x) {
        s1.push(x);
    }
    
    /** Removes the element from in front of queue and returns that element. */
    int pop() {
        int val, temp;
        while (!s1.empty()){
            val = s1.top(); s1.pop();
            s2.push(val);
        }
        s2.pop();
        while (!s2.empty()){
            temp = s2.top(); s2.pop();
            s1.push(temp);
        }
        return val;
    }
    
    /** Get the front element. */
    int peek() {
        int val, temp;
        while (!s1.empty()){
            val = s1.top(); s1.pop();
            s2.push(val);
        }
        while (!s2.empty()){
            temp = s2.top(); s2.pop();
            s1.push(temp);
        }
        return val;
        
    }
    
    /** Returns whether the queue is empty. */
    bool empty() {
        return (s1.empty());
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

